### PR TITLE
fix: status endpoint elements

### LINKED ISSUE #35: Fix 'status' endpoint
The `status` endpoint does not report any info, not even the build commit/time

### PR DESCRIPTION
**Pull request by xcoulon**
fixes #35

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: controller/status.go [hunk 1]
```diff
@@ -1,22 +1,94 @@
...
 import (
+	"fmt"
+
 	"github.com/fabric8-services/admin-console/app"
+	"github.com/fabric8-services/fabric8-common/log"
 	"github.com/goadesign/goa"
+	"github.com/jinzhu/gorm"
 )
 
+// StatusController implements the status resource.
```
Comment (dipak-pawar) [inline controller/status.go]: copy paste?

Comment (xcoulon) [reply controller/status.go]: yes :D

**Comment by xcoulon** [comment]
@dipak-pawar thanks for the review. Yes, I plan to add tests and fix the golangci issues as well.

**Comment by xcoulon** [comment]
@dipak-pawar @alexeykazakov so.... the build now runs `golangci-lint` aside of the hosted golangci (SaaS), and there are tests for the status controller 🎉
 👉please let me know if that's ok for you ;)

File: controller/status_blackbox_test.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+		config.IsDeveloperModeEnabledFunc = func() bool {
+			return true
+		}
+		config.DefaultConfigurationErrorFunc = func() error {
+			return nil
+		}
+		// when/then
+		apptest.ShowStatusServiceUnavailable(t, ctx, svc, ctrl)
+	})
+
```
Comment (alexeykazakov) [inline controller/status_blackbox_test.go]: Still missing a test with dev mode disabled and there are some configuration errors.
Also add tests to configuration to check all errors configuration can produce to make sure we catch every checked default value.

Comment (xcoulon) [reply controller/status_blackbox_test.go]: ok, I'll take care of these extra tests

**Comment by xcoulon** [comment]
ok, let me split the PR and move the Makefile changes in another one.

**Comment by xcoulon** [comment]
@alexeykazakov the PR is finally ready for another review of yours ;) Please let me know if the lack of tests has been sufficiently addressed

File: configuration/configuration_blackbox_test.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+			assert.True(t, config.IsDeveloperModeEnabled())
+		})
+	})
+
+}
+
+type envvars map[string]string
+
+// setenvs sets the given env vars and provides a function to unsetenvs them all in a `defer` call
+func setenvs(envs envvars) func() {
```
Comment (alexeykazakov) [inline configuration/configuration_blackbox_test.go]: Would be nice to move it to common. We use this setenvs function in a few repos. But we can do it in a separate PR.

File: controller/status_blackbox_test.go (hunk 2 above)
Comment (alexeykazakov) [reply controller/status_blackbox_test.go]: The configuration tests look good now but you forgot to add the test for status?

**Comment by xcoulon** [comment]
@alexeykazakov I added the missing status testcase in eb00227.

**Pull request merged**